src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```diff
@@ -0,0 +1,349 @@
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            InitHasher(Algorithm);
+        }
+
+        /// <summary>
+        /// ProcessRecord() override
+        /// Processing strings from pipeline
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            // For null input string the result hash is 'null'
+            if (InputString != null)
+            {
+                foreach (string str in InputString)
+                {
+                    byte[] bytehash = null;
+                    String hash = null;
+                    bytehash = hasher.ComputeHash(Encoding.UTF8.GetBytes(str));
+                    hash = BitConverter.ToString(bytehash).Replace("-","");
+                    WriteObject(hash);
+                }
+            }
+        }
+    }
+
+
+
+    /// <summary>
+    /// This class implements Get-FileHash
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "FileHash", DefaultParameterSetName = PathParameterSet, HelpUri = "https://go.microsoft.com/fwlink/?LinkId=517145")]
+    [OutputType(typeof(FileHashInfo))]
+    public class GetFileHashCommand : HashCmdletBase
+    {
+        /// <summary>
+        /// Path parameter
+        /// The paths of the files to calculate a hashs
+        /// Resolved wildcards
+        /// </summary>
+        /// <value></value>
+        [Parameter(Mandatory = true, ParameterSetName = PathParameterSet, Position = 0, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true)]
+        public String[] Path
+        {
+            get
+            {
+                return _paths;
+            }
+            set
+            {
+                _paths = value;
+            }
+        }
+
+        /// <summary>
+        /// LiteralPath parameter
+        /// The literal paths of the files to calculate a hashs
+        /// Don't resolved wildcards
+        /// </summary>
+        /// <value></value>
+        [Parameter(Mandatory = true, ParameterSetName = LiteralPathParameterSet, Position = 0, ValueFromPipelineByPropertyName = true)]
+        [Alias("PSPath")]
+        public String[] LiteralPath
+        {
+            get
+            {
+                return _paths;
+            }
+            set
+            {
+                _paths = value;
+            }
+        }
+
+        private String[] _paths;
+
+        /// <summary>
+        /// InputStream parameter
+        /// The stream of the file to calculate a hash
+        /// </summary>
+        /// <value></value>
+        [Parameter(Mandatory = true, ParameterSetName = StreamParameterSet, Position = 0)]
+        public Stream InputStream { get; set; }
+
+        /// <summary>
+        /// BeginProcessing() override
+        /// This is for hash function init
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            InitHasher(Algorithm);
+
+            if (ParameterSetName == StreamParameterSet)
```

Comment:
Oh, sorry. I remember your earlier comment about this, but all mixed up :-)
Fixed.